awx/ui_next/src/screens/Application/ApplicationAdd/ApplicationAdd.test.jsx

```diff
@@ -0,0 +1,190 @@
+      initialEntries: ['/applications/add'],
+    });
+    ApplicationsAPI.readOptions.mockResolvedValue(options);
+
+    ApplicationsAPI.create.mockResolvedValue({ data: { id: 8 } });
+    await act(async () => {
+      wrapper = mountWithContexts(<ApplicationAdd />, {
+        context: { router: { history } },
+      });
+    });
+
+    await act(async () => {
+      wrapper.find('input#name').simulate('change', {
+        target: { value: 'new foo', name: 'name' },
+      });
+      wrapper.find('input#description').simulate('change', {
+        target: { value: 'new bar', name: 'description' },
+      });
+      wrapper
+        .find('AnsibleSelect[name="authorization_grant_type"]')
+        .prop('onChange')({}, 'authorization code');
+
+      wrapper.find('input#redirect_uris').simulate('change', {
+        target: { value: 'https://www.google.com', name: 'redirect_uris' },
+      });
+      wrapper.find('AnsibleSelect[name="client_type"]').prop('onChange')(
+        {},
+        'confidential'
+      );
+      wrapper.find('OrganizationLookup').invoke('onChange')({
+        id: 1,
+        name: 'organization',
+      });
+    });
+
+    wrapper.update();
+    expect(wrapper.find('input#name').prop('value')).toBe('new foo');
+    expect(wrapper.find('input#description').prop('value')).toBe('new bar');
+    expect(wrapper.find('InnerChipGroup').length).toBe(1);
+    expect(wrapper.find('InnerChipGroup').text()).toBe('organization');
+    expect(
+      wrapper
+        .find('AnsibleSelect[name="authorization_grant_type"]')
+        .prop('value')
+    ).toBe('authorization code');
+    expect(
+      wrapper.find('AnsibleSelect[name="client_type"]').prop('value')
+    ).toBe('confidential');
+    expect(wrapper.find('input#redirect_uris').prop('value')).toBe(
+      'https://www.google.com'
+    );
+    await act(async () => {
+      wrapper.find('Formik').prop('onSubmit')({
+        authorization_grant_type: 'authorization-code',
+        client_type: 'confidential',
+        description: 'bar',
+        name: 'foo',
+        organization: { id: 1 },
+        redirect_uris: 'http://www.google.com',
+      });
+    });
+
+    expect(ApplicationsAPI.create).toBeCalledWith({
+      authorization_grant_type: 'authorization-code',
+      client_type: 'confidential',
+      description: 'bar',
+      name: 'foo',
+      organization: 1,
+      redirect_uris: 'http://www.google.com',
+    });
+    expect(history.location.pathname).toBe('/applications/8/details');
+  });
+
+  test('should cancel form properly', async () => {
+    const history = createMemoryHistory({
+      initialEntries: ['/applications/add'],
+    });
+    ApplicationsAPI.readOptions.mockResolvedValue(options);
+
+    ApplicationsAPI.create.mockResolvedValue({ data: { id: 8 } });
+    await act(async () => {
+      wrapper = mountWithContexts(<ApplicationAdd />, {
+        context: { router: { history } },
+      });
+      await act(async () => {
```

Comment:
should these `act` calls be nested? Never seen that before.

Reply:
No, they shoouldn't be